ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/DefaultSchemaInjector.java

```diff
@@ -213,28 +214,36 @@ private static boolean shouldInferSchema(
      *  1. If schema id is provided, format must support schema inference
      */
 
+    final String formatProp = isKey ? CommonCreateConfigs.KEY_FORMAT_PROPERTY
+        : CommonCreateConfigs.VALUE_FORMAT_PROPERTY;
+    final String schemaIdName =
+        isKey ? CommonCreateConfigs.KEY_SCHEMA_ID : CommonCreateConfigs.VALUE_SCHEMA_ID;
+    final String formatPropMsg = String.format("%s should support schema inference when %s is "
+        + "provided. Current format is %s.", formatProp, schemaIdName, formatInfo.getFormat());
+
+    final Format format;
+    try {
+      format = FormatFactory.of(formatInfo);
+    } catch (KsqlException e) {
+      if (e.getMessage().contains("does not support the following configs: [schemaId]")) {
+        throw new KsqlException(formatPropMsg);
+      }
```

Comment:
Because before https://github.com/confluentinc/ksql/pull/8401 is merged, we don't put `schemaId` to format properties and `FormatFactory.of()` doesn't throw. 

https://github.com/lihaosky/ksql/blob/305ae88b0ffd2c34be8e1f0f0f18dc80f5f3d40a/ksqldb-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceProperties.java#L190-L193

> A typo in the future this condition to throw a different exception

We have unit test covering this case if it helps. Otherwise, it's hard to check `FormatFactory.of` throws because of schema id problem... Unless we create different exception type?